src/analysis/normed_space/star/basic.lean

```diff
@@ -108,20 +108,14 @@ variables [normed_ring E] [star_ring E] [cstar_ring E]
 /-- In a C*-ring, star preserves the norm. -/
 @[priority 100] -- see Note [lower instance priority]
 instance to_normed_star_monoid : normed_star_monoid E :=
-⟨begin
-  intro x,
-  by_cases htriv : x = 0,
-  { simp only [htriv, star_zero] },
-  { have hnt : 0 < ∥x∥ := norm_pos_iff.mpr htriv,
-    have hnt_star : 0 < ∥x⋆∥ :=
-      norm_pos_iff.mpr ((add_equiv.map_ne_zero_iff star_add_equiv).mpr htriv),
-    have h₁ := calc
-      ∥x∥ * ∥x∥ = ∥x⋆ * x∥        : norm_star_mul_self.symm
-            ... ≤ ∥x⋆∥ * ∥x∥      : norm_mul_le _ _,
-    have h₂ := calc
-      ∥x⋆∥ * ∥x⋆∥ = ∥x * x⋆∥      : by rw [←norm_star_mul_self, star_star]
-             ... ≤ ∥x∥ * ∥x⋆∥     : norm_mul_le _ _,
-    exact le_antisymm (le_of_mul_le_mul_right h₂ hnt_star) (le_of_mul_le_mul_right h₁ hnt) },
+⟨λ x, begin
+  have n_le : ∀ {x : E}, ∥x⋆∥ ≤ ∥x∥,
+  { intros y,
+    by_cases y0 : y = 0,
+    { simp [y0] },
+    { exact le_of_mul_le_mul_right (by simp [← norm_star_mul_self, norm_mul_le])
+        (norm_pos_iff.mpr (star_ne_zero.mpr y0)) } },
+  exact n_le.antisymm ((congr_arg _ (star_star _)).ge.trans n_le)
```

Comment:
Or without `simp`:
```suggestion
⟨λ x, begin
  have n_le : ∀ {y : E}, ∥y∥ ≤ ∥y⋆∥,
  { intros y,
    by_cases y0 : y = 0,
    { rw [y0, star_zero] },
    { refine le_of_mul_le_mul_right _ (norm_pos_iff.mpr y0),
      exact (norm_star_mul_self.symm.trans_le $ norm_mul_le _ _) } },
  exact (n_le.trans_eq $ congr_arg _ $ star_star _).antisymm n_le,
```

(swapping the direction of `n_le` makes the proof easier)